server/src/test/java/io/crate/planner/operators/HashJoinTest.java

```diff
@@ -0,0 +1,71 @@
+package io.crate.planner.operators;
+
+import static io.crate.testing.Asserts.isReference;
+
+import java.util.List;
+import java.util.Map;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.crate.analyze.relations.AnalyzedRelation;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.RelationName;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SqlExpressions;
+import io.crate.testing.T3;
+
+public class HashJoinTest extends CrateDummyClusterServiceUnitTest {
+
+    private SqlExpressions sqlExpressions;
+    private AnalyzedRelation tr1;
+    private AnalyzedRelation tr2;
+    private AnalyzedRelation tr3;
+
+    @Before
+    public void prepare() throws Exception {
+        Map<RelationName, AnalyzedRelation> sources = T3.sources(clusterService);
+        sqlExpressions = new SqlExpressions(sources);
+        tr1 = sources.get(T3.T1);
+        tr2 = sources.get(T3.T2);
+        tr3 = sources.get(T3.T3);
+    }
+
+    @Test
+    public void test_create_lsh_rhs_hash_symbols_from_single_eq_condition() {
+        Symbol joinCondition = sqlExpressions.asSymbol("t1.a = t2.b");
+        var result = HashJoin.createHashSymbols(
+            List.of(tr1.relationName()),
+            List.of(tr2.relationName()),
+            joinCondition);
+
+        Assertions.assertThat(result.lhsHashSymbols()).satisfiesExactly(isReference("a"));
```

Comment:
Can we use isSQL() so that we can assert the fully qualified name? or add some new method in SymbolAssert like: isReferenceWithFqn() and use the referenceIdent to get the fqn.

For the next tests it can be dangerous to hide regressions if the same reference a is extracted for example.